ui/src/components/Table/DataTable.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  ColumnDef,
+  SortingState,
+  VisibilityState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { useState } from "react";
+
+import {
+  DropdownMenu,
+  DropdownMenuCheckboxItem,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+  Button,
+  Input,
+} from "@flow/components";
+import { useT } from "@flow/providers";
+
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from ".";
+
+interface DataTableProps<TData, TValue> {
+  columns: ColumnDef<TData, TValue>[];
+  data: TData[];
+  selectColumns?: boolean;
+  showFiltering?: boolean;
+}
```

Comment:
Is there a Prop directly useable in react-table?

Reply:
Explained here: https://github.com/reearth/reearth-flow/pull/161#discussion_r1602678773